### DISCUSSION TITLE
Make a PGN move given a FEN string

### DISCUSSION THREAD

**Discussion by ghost** [mliebelt/pgn-viewer#486] [Q&A]
Hi there, 

[chesscoach.me/moves](https://chesscoach.me/moves) helps chess students understand why a move is good by entering a FEN string.

![screencapture-chesscoach-me-moves-2024-01-02-20_59_01](https://github.com/mliebelt/PgnViewerJS/assets/4062000/bab1f3b1-ef11-429c-bcc4-f5f6560b2ac5)

As you can see in the image attached, the chessboard image is being generated on the server side with the help of [PHP Chess](https://github.com/chesslablab/php-chess). I'd want to use the PgnViewerJs script instead. I was wondering if it could be initialized to make a PGN move from a particular FEN position rather than from the start position.

Thanks for the help,

**Comment by mliebelt** [comment]
Well, that should be easy. PgnViewerJS has 4 different modes, one of them is `pgnBoard`, that just shows a board.

The following examples shows, what parameter is needed in the configuration: https://mliebelt.github.io/PgnViewerJS/examples.html#1022

There are many more examples there, so you can of course select the theme, pieceStyle, and many more configuration parameters. To set the position, the parameter is `position` or `fen`.

**Comment by ghost** [comment]
Thank you @mliebelt for the examples. 

Let me ask one last question, how can the chessboard images be downloaded? 

Keep it up,

**Comment by mliebelt** [comment]
What do you mean by "chessboard images"? The whole board is completely built in HTML, the images are the ones of the background, the pieces, some marks, ... It is completely rendered in the browser, so there is no (easy) way to have the complete board as an image.

**Comment by ghost** [comment]
Something like right-click on the image and then click on Save image as...

![saveimage](https://github.com/mliebelt/PgnViewerJS/assets/4062000/c8c16794-303f-4961-bef1-3723c6598df4)

**Reply by mliebelt** [reply]
I do understand the idea, but I say that the working is quite different, so you don't see on the UI one image, but a multitude of images that combined give the board. I checked if there could be an easy approach to do that, and got something that may work. It is outside the scope of this library to add something like that, in my opinion.

The base idea is the following:

* Use a library that uses a canvas to get the image as URL.
* Save that URL then as a file.

The code would be something like (not tried, needs tuning).

```Javascript
// Get div element
const div = document.getElementById('divToSave'); // This would be the div for the board

// Use html2canvas, library has to be installed/available
html2canvas(div).then(canvas => {

  // Get image data URL
  const imgDataURL = canvas.toDataURL();

  // Save image data URL to file  
  const a = document.createElement('a');
  a.href = imgDataURL; 

  // Set download attribute to filename
  a.download = 'board.png';

  // Click the anchor to trigger download
  a.click(); 
});```

I hope that idea helps to implement that by yourself.
